code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,305 @@
+			myparent.onEliteWon()
+
+/mob/living/simple_animal/hostile/asteroid/elite/death()
+	. = ..()
+	if(myparent)
+		myparent.onEliteLoss()
+
+//The Glowing Crevice, the actual "spawn-point" of elites, handles the spawning, arena, and procs for dealing with basic scenarios.
+
+/obj/structure/elite_crevice
+	name = "glowing crevice"
+	desc = "A glowing, red hole which doesn't seem to have a bottom.  You feel pressured to reach your hand out towards it..."
+	armor = list("melee" = 100, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
+	max_integrity = 10000
+	var/activity = 0
+	var/boosted = 0
+	var/times_won = 0
+	var/mob/living/carbon/human/activator = null
+	var/mob/living/simple_animal/hostile/asteroid/elite/mychild = null
+	var/potentialspawns = list(/mob/living/simple_animal/hostile/asteroid/elite/goliath)
+	icon = 'icons/mob/lavaland/elite_lavaland_monsters.dmi'
+	icon_state = "elite_crevice"
+	light_color = LIGHT_COLOR_RED
+	light_range = 3
+	anchored = TRUE
+	density = FALSE
+	
+/obj/structure/elite_crevice/attack_hand(mob/user)
+	if(activity == 0)
+		if(istype(user, /mob/living/carbon/human))
+			activity = 1
+			var/mob/dead/observer/elitemind = null
+			visible_message("<span class='boldwarning'>The crevice glows.  Your instincts tell you to step back.</span>")
+			activator = user
+			if(boosted == 1)
+				visible_message("<span class='boldwarning'>Something within the crevice stirs...</span>")
+				var/list/candidates = pollCandidatesForMob("Do you want to play as a lavaland elite?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, src, POLL_IGNORE_SENTIENCE_POTION)
+				if(candidates.len)
+					visible_message("<span class='boldwarning'>The stirring sounds increase in volume!</span>")
+					elitemind = pick(candidates)
+					SEND_SOUND(elitemind, sound('sound/effects/magic.ogg'))
+					to_chat(elitemind, "<b>You have been chosen to play as a Lavaland Elite.</b>")
+					to_chat(elitemind, "<b>In a few seconds, you will be summoned on Lavaland as a monster to fight your activator, in a fight to the death.</b>")
+					to_chat(elitemind, "<b>Your attacks can be switched using the buttons on the top left of the HUD, and used by clicking on targets or tiles similar to a gun.</b>")
+					to_chat(elitemind, "<b>While the opponent might have an upper hand with  powerful mining equipment and tools, you have great power normally limited by AI mobs.</b>")
+					to_chat(elitemind, "<b>If you want to win, you'll have to use your powers in creative ways to ensure the kill.  It's suggested you try using them all as soon as possible.</b>")
+					to_chat(elitemind, "<b>Should you win, you'll receive extra information regarding what to do after.  Good luck!</b>")
+					sleep(100)
+				else
+					visible_message("<span class='boldwarning'>The stirring stops, and nothing emerges.  Perhaps try again later.</span>")
+					activity = 0
+					activator = null
+					return
+			else
+				sleep(30)
```

Comment:
timer

Reply:
All sleeps are now timers in upcoming fix.